librad/src/net/peer.rs

```diff
@@ -436,43 +436,47 @@ where
 
         match has.urn.proto {
             uri::Protocol::Git => {
+                let peer_id = has.origin.clone().unwrap_or_else(|| provider.clone());
+                let is_tracked = match self.inner.lock().unwrap().is_tracked(&has.urn, &peer_id) {
+                    Ok(b) => b,
+                    Err(e) => {
+                        tracing::error!(err = %e, "Git::Storage::is_tracked error");
+                        return PutResult::Error;
+                    },
+                };
                 let res = match has.rev {
                     // TODO: may need to fetch eagerly if we tracked while offline (#141)
-                    None => PutResult::Uninteresting,
-                    Some(Rev::Git(head)) => {
+                    Some(Rev::Git(head)) if is_tracked => {
```

Comment:
I think we can actually fix the TODO: if we only check `is_tracked`, we'll also fetch if we haven't yet fetched any data from the tracked peer.

Reply:
Addressing this in a follow up PR.